src/prefect/cli/server.py

```diff
@@ -264,58 +264,74 @@ async def start(
-        process = await anyio.open_process(
-            command=command,
-            env=server_env,
-        )
-
-        process_id = process.pid
-        if background:
-            await pid_file.write_text(str(process_id))
-
-            app.console.print(
-                "The Prefect server is running in the background. Run `prefect"
-                " server stop` to stop it."
-            )
-            return
-
-        async with process:
-            # Explicitly handle the interrupt signal here, as it will allow us to
-            # cleanly stop the uvicorn server. Failing to do that may cause a
-            # large amount of anyio error traces on the terminal, because the
-            # SIGINT is handled by Typer/Click in this process (the parent process)
-            # and will start shutting down subprocesses:
-            # https://github.com/PrefectHQ/server/issues/2475
+    if background:
+        _run_in_background(pid_file, server_settings, host, port, keep_alive_timeout)
+    else:
+        _run_in_foreground(pid_file, server_settings, host, port, keep_alive_timeout)
+
+
+def _run_in_background(
+    pid_file: Path,
+    server_settings: dict[str, str],
+    host: str,
+    port: int,
+    keep_alive_timeout: int,
+) -> None:
+    command = [
+        sys.executable,
+        "-m",
+        "uvicorn",
+        "--app-dir",
+        str(prefect.__module_path__.parent),
+        "--factory",
+        "prefect.server.api.server:create_app",
+        "--host",
+        str(host),
+        "--port",
+        str(port),
+        "--timeout-keep-alive",
+        str(keep_alive_timeout),
+    ]
+    logger.debug("Opening server process with command: %s", shlex.join(command))
+    process = subprocess.Popen(
+        command,
+        env={
+            **os.environ,
+            **server_settings,
+            "PREFECT__SERVER_FINAL": "1",
+        },
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
 
-            setup_signal_handlers_server(
-                process_id, "the Prefect server", app.console.print
-            )
+    process_id = process.pid
+    pid_file.write_text(str(process_id))
 
-            await consume_process_output(process, sys.stdout, sys.stderr)
+    app.console.print(
+        "The Prefect server is running in the background. Run `prefect"
+        " server stop` to stop it."
+    )
 
-    except anyio.EndOfStream:
-        logging.error("Subprocess stream ended unexpectedly")
 
-    # This should only be reached if the server crashed, or was forcibly terminated,
-    # hence we exit with an error
-    exit_with_error("Server stopped!", code=process.returncode)
+def _run_in_foreground(
+    pid_file: Path,
+    server_settings: dict[str, str],
+    host: str,
+    port: int,
+    keep_alive_timeout: int,
+) -> None:
+    from prefect.server.api.server import create_app
+
+    with temporary_settings(
+        {getattr(prefect.settings, k): v for k, v in server_settings.items()}
```

Comment:
Is this cheeseball?  Should I just explicitly set the Prefect settings here?  I couldn't decide...

Reply:
This seems reasonable to me, considering the current state of settings. We need to update `temporary_settings` to be more ergonomic now we're using pydantic settings, but this is the right way to do it now.